Realm/RLMUser.h

```diff
@@ -114,6 +114,8 @@ NS_ASSUME_NONNULL_BEGIN
  @param partitionValue The `RLMBSON` value the Realm is partitioned on.
  @param clientResetMode Determines file recovery behavior in the event of a client reset.
                         See: https://docs.mongodb.com/realm/sync/error-handling/client-resets/
+ 
+ @return A default configuration object with the sync configuration set to use the given partition value.
```

Comment:
We should avoid using the word `default` here as we have a concept of a "default configuration" that this doesn't interact with.